.github/workflows/create-release-branch.yml

```diff
@@ -0,0 +1,76 @@
+
+on:
+  workflow_dispatch:
+    inputs:
+      patch_version:
+        description: Patch version (integer)
+        required: true
+env:
+  NODE_ENV: production
+  RELEASE_VERSION: '${{ format(''{0}.{1}'', github.ref_name, github.event.inputs.patch_version) }}'
+
+jobs:
+  create_release:
+    name: Create release branch
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Check branch
+        run: echo "${{ github.ref_name }}" | grep -Pq '^v3\.\d+$'
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: true
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~1.17'
+      - name: Download Go tool dependencies
+        run: |
+          cd tools
+          go mod download
+      - name: Initialize Go build cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-${{ github.ref }}
+          restore-keys: |
+            ${{ runner.os }}-go-build-refs/heads/v
+      - name: Initialize tool binary cache
+        id: tools-cache
+        uses: actions/cache@v2
+        with:
+          path: tools/bin
+          key: ${{ runner.os }}-tools-${{ hashFiles('tools/**') }}
+      - name: Make Mage
+        run: make tools/bin/mage
+        if: steps.tools-cache.outputs.cache-hit != 'true'
+      - name: Install JS dependencies
+        run: tools/bin/mage js:deps
+      - name: Update version information
+        run: echo "${RELEASE_VERSION}" | tools/bin/mage version:files
+      - name: Update changelog
+        run: 'cat CHANGELOG.md | tr ''\n'' ''\000'' |
+          sed ''s/## [A-Z][a-z]*\x00\x00#/\x01\x00#/g''|
+          sed ''s/## \[Unreleased\]/## \[Unreleased\]\x00\x00### Added\x00\x00### Changed\x00\x00### Deprecated\x00\x00### Removed\x00\x00### Fixed\x00\x00### Security\x00\x00''"## \\[${RELEASE_VERSION#v}\\] - $(date +%Y-%m-%d)"''/'' |
+          tr ''\000'' ''\n'' |
+          sed ''/\x01/d'' |
+          sed "s/\[unreleased\]: https:\/\/github.com\/$(echo "${GITHUB_REPOSITORY}" | sed ''s/\//\\\//g'')\/compare\/\(v3\.[0-9][0-9]*\.[0-9][0-9]*\)\.\.\.\(v3\.[0-9][0-9]*\)/[unreleased]: https:\/\/github.com\/$(echo "${GITHUB_REPOSITORY}" | sed ''s/\//\\\//g'')\/compare\/${RELEASE_VERSION}...${GITHUB_REF_NAME}\n[${RELEASE_VERSION#v}]: https:\/\/github.com\/$(echo "${GITHUB_REPOSITORY}" | sed ''s/\//\\\//g'')\/compare\/\1...${RELEASE_VERSION}/" > CHANGELOG2.md
+          && mv -f CHANGELOG2.md CHANGELOG.md
+          && git add CHANGELOG.md'
+      - name: Update submodules
+        id: update_submodules
+        run: |
+          git submodule update --init --remote
+      - name: Prepare release branch
+        run: |
+          git config user.name "The Things Bot"
+          git config user.email "github@thethingsindustries.com"
+          git checkout -b "release/${RELEASE_VERSION}"
+          git add .
```

Comment:
From design perspective it's kinda difficult, because we delegate certain things to commands. "mage, please update version files, I don't care how you do it, just do it".

Since we always execute the same few commands, it's not like much can go wrong.

Reply:
I don't know. The way I see it, we know what _should_ change, so we can just explicitly add those files, being confident that we're not committing unexpected changes.

@adriansmares what do you think?